File: starters/react-native/src/push/push-hook.ts
export const PUSH_HOOK_CONTRACT_VERSION = 'push-hook/2.0.0';

export interface PushHook {
  readonly contractVersion: typeof PUSH_HOOK_CONTRACT_VERSION;
  readonly enabled: boolean;
  register(): Promise<'disabled' | 'registered'>;
  unregister(): Promise<void>;
}

export function createDisabledPushHook(): PushHook {
  return Object.freeze({
    contractVersion: PUSH_HOOK_CONTRACT_VERSION,
    enabled: false,
    register: async (): Promise<'disabled'> => 'disabled',
    unregister: async () => undefined,
  });
}

File: starters/flutter/lib/src/core/push/push_hook.dart
const pushHookContractVersion = 'push-hook/2.0.0';

abstract interface class PushHook {
  String get contractVersion;
  bool get enabled;
  Future<String> register();
  Future<void> unregister();
}

final class DisabledPushHook implements PushHook {
  const DisabledPushHook();

  @override
  String get contractVersion => pushHookContractVersion;

  @override
  bool get enabled => false;

  @override
  Future<String> register() async => 'disabled';

  @override
  Future<void> unregister() async {}
}
